src/citrine/resources/file_link.py

```diff
@@ -883,14 +888,30 @@ def ingest(self,
         """
         from citrine.resources.ingestion import IngestionCollection
 
-        targets = [self._resolve_file_link(f) for f in files]
+        def resolve_local_paths(candidate: Union[FileLink, Path, str]) -> Union[FileLink, str]:
+            """Convert a string or Path into a FileLink if it resolves locally."""
+            if not isinstance(candidate, FileLink):
+                try:
+                    resolved = Path(candidate).resolve(strict=True)
+                    return FileLink(filename=resolved.name, url=resolved.as_uri())
+                except FileNotFoundError as e:
+                    if isinstance(candidate, Path):  # Unresolvable path passed
+                        raise e
+
+            return candidate  # It was a FileLink or an unresolvable string
+
+        if upload:
+            targets = [self._resolve_file_link(resolve_local_paths(f)) for f in files]
+        else:
+            targets = [self._resolve_file_link(f) for f in files]
         offplatform = [f for f in targets if not self._is_on_platform_url(f.url)]
+
         if upload:
             with TemporaryDirectory() as downloads:
                 paths = [Path(downloads) / f.filename for f in offplatform]
                 for file_link, path in zip(offplatform, paths):
                     self.download(file_link=file_link, local_path=path)
-                onplatform = [self.upload(file_path=p) for p in paths]
+                onplatform = [self.upload(file_path=p, dest_name=p.name) for p in paths]
```

Comment:
just curious what happens with duplicate paths. and duplicate names.

Reply:
Yeah, duplicate filenames would cause a mess.  I'll clean that up.  :bow: